asdf/_tests/test_info.py

```diff
@@ -633,6 +633,9 @@ def __init__(self):
     def __asdf_traverse__(self):
         return {"the_meaning": self.the_meaning, "clown": self.clown, "recursive": self.recursive}
 
+    def __str__(self):
+        return "rec ref"
```

Comment:
This was added because with this PR (and without this change) the assert in `test_recursive_info_object_support`:
https://github.com/asdf-format/asdf/blob/ecb4d4f6308738c4de0da8eec5bd272e90f3954e/asdf/_tests/test_info.py#L650
would fail due to the output line being truncated due to this test object (`RecursiveObjectWithInfoSupport`) not having a `__str__`.